### ISSUE TITLE
[Enhancement]: Unique constrain with multiple fields

### ISSUE THREAD

**Issue by ljia387** [milvus-io/milvus#36704]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

I would like to request a enhancement to support Unique constrain with multiple fields

### Why is this needed?

After introduce chunking, a record is unique with doc id and chunk id.  It seems that milvus will support primary key deduplication in 3.0,   please also consider apply unique constrain on multiple fields instead of single field.

**Comment by xiaofan-luan** [comment]
I guess what we need is a composite primary key. 

This is not the goal for milvus 3.0 right now but you can probably combine a varchar key with "docid_chunkid" as a work round .

**Comment by xiaofan-luan** [comment]
But yes, composite key will be a great feature to have

**Comment by ljia387** [comment]
> I guess what we need is a composite primary key.
> 
> This is not the goal for milvus 3.0 right now but you can probably combine a varchar key with "docid_chunkid" as a work round .

Yes, combine varchar can be a wordaround. But it's better we can have this composite primiary key supoortyed by nature in VDB.

**Comment by xiaofan-luan** [comment]
Tracked as #52600 — composite primary key with an order-preserving concatenated encoding and a per-field sort direction. The separator-collision problem with the string-concat workaround suggested here is resolved by construction via length-prefixed encoding.
